Lib/locale.py

```diff
@@ -633,6 +633,7 @@ def getpreferredencoding(do_setlocale = True):
                 if not result and sys.platform == 'darwin':
```

Comment:
I propose to replace "if ...: ... return result else: ... return result" with "if ...: ... else ...; return result", just factorize the code which is in common. That's all.